storage/innobase/include/rem0rec.inl

```diff
@@ -156,6 +156,9 @@ rec_set_bit_field_1(
 	ut_ad(((mask >> shift) << shift) == mask);
 	ut_ad(((val << shift) & mask) == (val << shift));
 
+#ifndef DBUG_OFF
+	MEM_MAKE_DEFINED(rec - offs, 1);
+#endif
 	mach_write_to_1(rec - offs,
 			(mach_read_from_1(rec - offs) & ~mask)
 			| (val << shift));
```

Comment:
This looks incorrect to me. Why would we claim that all bits at `rec[-offs]` are initialized when we are only overwriting some of the bits here? What would fail if this change and the similar change to `rec_set_bit_field_2()` were omitted?

Reply:
The trace was like in MDEV-36316.

I agree it seem overly incorrect claiming all bits are initialised. It only occurred in Debug mode so I'm assuming a less optimised code makes this look different.  I'll look at forcing a higher optimisation on these blocks as an alternative.